Automattic/pocket-casts-android
# Adding Showkase to manage our Compose components

## Description

This introduces the Showkase tool to help us find and reuse existing Compose components.
https://github.com/airbnb/Showkase

I have yet to add all the components and instead marked the previews as private to get it set up with only a few initially. 

It would be good to write new Compose pages using components but if they don't exist then create new components. Having the `@Preview` annotation adds the component to Showkase but I think the best way might be to do the following. Adding a light and dark preview.

```
@ShowkaseComposable(name = "RowButton", group = "Button", styleName = "Light", defaultStyle = true)
@Preview(name = "Light")
@Composable
fun RowButtonsLightPreview() {
    AppThemeWithBackground(Theme.ThemeType.LIGHT) {
        RowButton(text = "Accept", onClick = {})
    }
}

@ShowkaseComposable(name = "RowButton", group = "Button", styleName = "Dark")
@Preview(name = "Dark")
@Composable
fun RowButtonsDarkPreview() {
    AppThemeWithBackground(Theme.ThemeType.DARK) {
        RowButton(text = "Accept", onClick = {})
    }
}
```

The colour of the developer icon isn't right but I might rewrite the page in Compose.

## Testing Instructions
1. Tap the Profile tab
2. Tap the settings cog
3. Tap the Developer option
4. Tap Showkase
5. ✅ Verify you can see the components

## Screenshots

<img  width="300" src="https://github.com/Automattic/pocket-casts-android/assets/308331/f0ad7f30-4482-4fd2-ba3a-437d99ce9f54" />
<img  width="300" src="https://github.com/Automattic/pocket-casts-android/assets/308331/a0060df8-5679-4558-8da6-a774f8f5c93d" />
<img  width="300" src="https://github.com/Automattic/pocket-casts-android/assets/308331/268f3766-b052-489b-8294-9de3519361ca" />
